tools/run_tests/run_interop_tests.py

```diff
@@ -843,7 +902,7 @@ def aggregate_http2_results(stdout):
     for language in languages:
       for test_case in _TEST_CASES:
         if not test_case in language.unimplemented_test_cases():
-          if not test_case in skip_server:
+          if not test_case in skip_server and not args.http2_badserver_interop:
```

Comment:
Moved the if condition several lines up. This should be executed even when http2_interop is true as it is launching servers which are used during http2_interop.